openedx/adg/lms/applications/managers.py

```diff
@@ -50,9 +50,17 @@ def business_line_and_common_business_line_prereq_course_groups_for_user(self, u
         Returns:
             list: List of business line prerequisite course groups for a user
         """
-        return self.get_queryset().filter(
-            Q(is_common_business_line_prerequisite=True) | Q(business_line_prerequisite=user.application.business_line)
-        )
+        if user:
+            course_groups = self.get_queryset().filter(
+                Q(is_common_business_line_prerequisite=True) | Q(
+                    business_line_prerequisite=user.application.business_line
+                )
+            )
+        else:
+            course_groups = self.get_queryset().filter(
+                Q(is_common_business_line_prerequisite=True) | Q(business_line_prerequisite__isnull=False)
+            )
+        return course_groups
```

Comment:
Why can't we create two separate functions, instead of using these conditionals and changing the default value of `user` parameter to null? How about keep the existing function as it was, and create a new function for the else part, that does not accept any user?

Reply:
To avoid duplicity of code and as per code review suggestion, have modified the current function.